### DISCUSSION TITLE
Lighter customization

### DISCUSSION THREAD

**Discussion by krisbalintona** [alphapapa/hammy.el#28] [Ideas]
I am wondering whether it would be prudent to have more options for customizing the mode line lighter. Of course, users could always override/redefine `hammy-mode-lighter` to their tastes, but enough users might want such customization, so why not make it easier if possible? Here are a few comments on the matter:
+ I often like having elements of my mode line surrounded by parentheses or brackets; it helps for visual clarity. (If I remember correctly, the default for `org-clock`'s mode line string does this.)
+ A user might set `hammy-mode-lighter-prefix` to an empty string, but the colon that follows remains.
+ What if a user wants only the remaining duration without the name of the hammy or the title of the current interval? Perhaps a user only really ever has one timer open at a time, removing the necessity of disambiguating which duration belongs to which timer.

I imagine the above to be mostly pertaining to altering `hammy-mode-lighter` and adding a few defcustoms. However, I also have in mind possibilities that might require more extensive modifications:
+ Regarding having only the remaining duration of hammys shown: perhaps we can also let the user set the face for the remaining duration for each internal in their `hammy-define`s too (like one can for the name of the interval). (This would also be a means for disambiguating hammys without the need for their names and internal titles.)
+ Changing the format of the time shown in the mode line. Instead of remaining time and overdue time, I prefer an "ELAPSED/DURATION" format, and perhaps other users might prefer other formats too.

These are some thoughts after using hammy for a few weeks. I believe this to be a worthy topic of discussion because though this is a matter of aesthetics -- which might seem like a trivial topic -- aesthetics influence how one interacts with the tool and how useful the tool is for them. This is one reason why packages like writeroom-mode, darkroom, olivetti, etc. are so popular. More concretely, I find long elements in my mode-line (or tab-bar, via `tab-bar-format`) a distraction. I do not believe these points to be more important than the functionality of your package (which is well-reasoned and useful), but think they nevertheless merit discussion.

**Comment by alphapapa** [comment]
Sure, we could probably use `format-spec` to make it more easily customized.

I probably won't have time to work on this soon, so patches welcome.  Thanks.

**Reply by krisbalintona** [reply]
No problem. I agree that this isn't a high priority task. If I write some tweaks that work for me I'll try putting them into a PR. Thanks for your input.
